#28 Updates names and adds descriptions

shabbyjoon opened the pull request.

jugglinmike commented on frontend/settings.jsx:
```diff
@@ -40,18 +40,26 @@ export default function Settings({consumerName, producerName, assignmentName}) {
-                <legend>{assignmentName} schema</legend>
+                <legend>{assignmentName} Table</legend>
+
+                <FormField
```
The `FormField` component isn't exactly what we want here because it will create an HTML `<label>` element without a corresponding `<input>`. That may confuse screen readers and/or their users.

A `<p>` tag should be fine for our purposes. Even better would be to extend the containing `<fieldset>` with an `aria-describedby` attribute which references the `<p>`. Something like:

```html
<fieldset style={{marginBottom: '1em'}} aria-describedby="deliveries-description">
    <legend>{assignmentName} Table</legend>

    <p
        style={{width: '80%'}}
        id="deliveries-description">
        Specify the table where you want your planned deliveries to go.
        You will need to specify the column names in your table so that your planning data
        is populated correctly.
    </p>
```

jugglinmike commented on frontend/settings.jsx:
```diff
@@ -40,18 +40,26 @@ export default function Settings({consumerName, producerName, assignmentName}) {
+                <FormField
+                    style={{width: '80%'}}
+                    label="Specify the table where you want your planned delivieries to go. 
+                    You will need to specifiy the column names in your table so that your planning data
```
"delivieries" -> "deliveries"
"specifiy" -> "specify"

jugglinmike commented on frontend/settings.jsx:
```diff
@@ -104,7 +112,14 @@ export default function Settings({consumerName, producerName, assignmentName}) {
+                <legend>{consumerName} Table</legend>
+
+                <FormField
+                    style={{width: '80%'}}
```
This component can get pretty wide--wide enough to make reading a paragraph of text awkward. My guess is that this width is intended to improve readability in that case. If that's right, then I'd recommend a different approach: `maxWidth: '40em'`. Here's why:

- By expressing the constrain in terms of `em`, it will be enforced according to the size of the characters instead of the width of the container. That way, if someone changes the character size, they won't have to update this value. This will also serve folks who use very wide monitors (for them, "80% of the parent" might result in lines that are still too long for comfort).
- By using `maxWidth` instead of `width`, the text can use all of the available space when space is limited.

shabbyjoon replied:
Thank you!

jugglinmike commented:
This is looking good, @shabbyjoon! There are just a couple things left, but since I was having trouble explaining them, I thought I'd make the changes and see what you think:

- the HTML `id` attribute has to be unique for the whole page, so I've changed it to something different from each of the `<p>` tags
- screen readers only recognize descriptions when they're declared with the `aria-describedby` attribute. I've added that to each `<fieldset>` element. Now, they each reference the new `<p>` tags, so screen readers will describe each set of form fields by reading the text in the paragraph
- In JSX, the "single quote" character needs to be written with an HTML character code. I've replaced the two occurrences with `&apos;`

Do those changes look alright to you?

shabbyjoon commented:
> This is looking good, @shabbyjoon! There are just a couple things left, but since I was having trouble explaining them, I thought I'd make the changes and see what you think:
> 
> * the HTML `id` attribute has to be unique for the whole page, so I've changed it to something different from each of the `<p>` tags
> * screen readers only recognize descriptions when they're declared with the `aria-describedby` attribute. I've added that to each `<fieldset>` element. Now, they each reference the new `<p>` tags, so screen readers will describe each set of form fields by reading the text in the paragraph
> * In JSX, the "single quote" character needs to be written with an HTML character code. I've replaced the two occurrences with `&apos;`
> 
> Do those changes look alright to you?

I'm sorry I missed this! They do and thank you for the explanation!

jugglinmike commented:
Thanks for the patch!
